connectors/sources/microsoft_teams.py

```diff
@@ -266,6 +627,84 @@ def client(self):
             tenant_id, client_id, client_secret, username, password
         )
 
+    async def _consumer(self):
+        """Async generator to process entries of the queue
+
+        Yields:
+            dictionary: Documents from Microsoft Teams.
+        """
+        while self.tasks > 0:
+            _, item = await self.queue.get()
+
+            if isinstance(item, EndSignal):
+                self.tasks -= 1
+            else:
+                yield item
+
+    def verify_filename_for_extraction(self, filename):
+        attachment_extension = list(os.path.splitext(filename))
+        if "" in attachment_extension:
+            self._logger.debug(
+                f"Files without extension are not supported, skipping {filename}."
+            )
+            return
+        if "." not in filename:
+            self._logger.debug(
+                f"Files without extension are not supported by TIKA, skipping {filename}."
+            )
+            return
+        if attachment_extension[-1].lower() not in TIKA_SUPPORTED_FILETYPES:
+            self._logger.debug(
+                f"Files with the extension {attachment_extension[-1]} are not supported, skipping {filename}."
+            )
+            return
+        return True
+
+    async def _download_content_for_attachment(self, download_func, original_filename):
+        attachment = None
+        source_file_name = ""
+
+        try:
+            async with NamedTemporaryFile(mode="wb", delete=False) as async_buffer:
+                source_file_name = async_buffer.name
+                await download_func(async_buffer)
+
+            self._logger.debug(
+                f"Download completed for file: {original_filename}. Calling convert_to_b64"
+            )
+            await asyncio.to_thread(
+                convert_to_b64,
+                source=source_file_name,
+            )
+            async with aiofiles.open(file=source_file_name, mode="r") as target_file:
+                # base64 on macOS will add a EOL, so we strip() here
+                attachment = (await target_file.read()).strip()
+
+        finally:
+            if source_file_name:
+                await remove(str(source_file_name))
+
+        return attachment
+
+    def map_document_with_schema(
+        self,
+        document,
+        item,
+        document_type,
+    ):
+        """Prepare key mappings for documents
+
+        Args:
+            document(dictionary): Modified document
+            item (dictionary): Document from Microsoft Teams.
+            document_type(string): Name of function to be called for fetching the mapping.
+
+        Returns:
+            dictionary: Modified document with the help of adapter schema.
+        """
+        for elasticsearch_field, sharepoint_field in document_type().items():
+            document[elasticsearch_field] = item[sharepoint_field]
```

Comment:
No change request, I just wanted to point out that this is a really neat way to deal with mappings. I will personally try to use this in the Salesforce connector sometime. Nice!

Reply:
Although one followup question: is it possible for `item[sharepoint_field]` to throw a `KeyError` or are we always sure that the API returns the expected response format?
If we aren't sure, using `item.get(sharepoint_field)` might be safer.

Reply:
I have already verified for the fields, they will come everytime in `item` but still we can use `.get` to make it more safe.